packages/taiko-client/proposer/proposer.go

```diff
@@ -294,6 +296,30 @@ func (p *Proposer) ProposeOp(ctx context.Context) error {
 
 // ProposeTxList proposes the given transactions lists to TaikoL1 smart contract.
 func (p *Proposer) ProposeTxLists(ctx context.Context, txLists []types.Transactions) error {
+	l2Head, err := p.rpc.L2.BlockNumber(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get L2 chain head number: %w", err)
+	}
+
+	// Check if the current L2 chain is after Pacaya fork, propose blocks batch.
+	if p.chainConfig.IsPacaya(new(big.Int).SetUint64(l2Head + 1)) {
+		preconfRouter, err := p.rpc.ResolvePacaya(nil, "preconf_router", true)
+		if err != nil {
+			return fmt.Errorf("failed to resolve preconfirmation router address: %w", err)
+		}
+
+		if preconfRouter != rpc.ZeroAddress {
+			log.Info("Preconfirmation router is set, skipping proposing blocks batch")
+			return nil
```

Comment:
If `preconfRouter != nill` means the proposer no longer needs to work anymore. Is it right?

Reply:
if preconf router is a non zero address then proposer should not propose

Reply:
Okay, I understand.